engines/libaio.c

```diff
@@ -85,6 +112,24 @@ static int fio_libaio_prep(struct thread_data fio_unused *td, struct io_u *io_u)
 	return 0;
 }
 
+static void fio_libaio_prio_prep(struct thread_data *td, struct io_u *io_u)
+{
+	struct libaio_options *o = td->eo;
+	if (o->cmdprio_percentage == 0)
+		return;
+	if (rand_between(&td->prio_state, 0, 99) < o->cmdprio_percentage) {
+		dprint(FD_IO, "Enable PRIO\n");
+		io_u->iocb.aio_reqprio = IOPRIO_CLASS_RT << IOPRIO_CLASS_SHIFT;
+		io_u->iocb.u.c.flags |= IOCB_FLAG_IOPRIO;
+		io_u->priority_bit = 1;
```

Comment:
Is this correct? From my quick look, you want (1U << 13) set, which means RT priority class. This gets translated into NCQ prio in libata at least, whereas any other class does not.

Reply:
I can see on the bus trace this sets the priority bit as desired, and it looks like `IOPRIO_CLASS_RT` is 1 and `IOPRIO_CLASS_SHIFT` is 13.

Reply:
Sorry was my being blind, thought the priority_bit was the field in the iocb. Yes, the other bit looks correct, you just want priority class RT set with no data value, which is 1U << 13, or CLASS_RT << CLASS_SHIFT as you did.

Reply:
But then that begs the question why you add priority_bit, instead of just using an io_u flag?

Reply:
Then you also don't have to clear it if prio isn't set.

Reply:
That's a good idea, I didn't think about adding a new io_u flag.

Reply:
I tried leaving the io_u->flags alone when priority is not set, but that seemed to leave the priority flag set after it was set once.